service/history/nDCWorkflowResetter.go

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+//go:generate mockgen -copyright_file ../../LICENSE -package $GOPACKAGE -source $GOFILE -destination nDCWorkflowResetter_mock.go
+
+package history
+
+import (
+	ctx "context"
+
+	"github.com/pborman/uuid"
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/definition"
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/log/tag"
+	"github.com/uber/cadence/common/persistence"
+)
+
+type (
+	nDCWorkflowResetterCompleteFn func()
+
+	nDCWorkflowResetter interface {
+		resetWorkflow(
+			ctx ctx.Context,
+			baseEventID int64,
+			baseVersion int64,
+		) (mutableState, nDCWorkflowResetterCompleteFn, error)
+	}
+
+	nDCWorkflowResetterImpl struct {
+		shard          ShardContext
+		transactionMgr nDCTransactionMgr
+		historyV2Mgr   persistence.HistoryV2Manager
+		stateRebuilder nDCStateRebuilder
+
+		domainID   string
+		workflowID string
+		baseRunID  string
+		newContext workflowExecutionContext
+		newRunID   string
+
+		logger log.Logger
+	}
+)
+
+var _ nDCWorkflowResetter = (*nDCWorkflowResetterImpl)(nil)
+
+func newNDCWorkflowResetter(
+	shard ShardContext,
+	transactionMgr nDCTransactionMgr,
+
```

Comment:
spurious newline.

Reply:
this is intentional, making the input variables easy to read